src/main/java/picard/illumina/parser/NewIlluminaDataProvider.java

```diff
@@ -0,0 +1,116 @@
+package picard.illumina.parser;
+
+import picard.illumina.BarcodeMatch;
+import picard.illumina.BarcodeMetric;
+import picard.illumina.parser.readers.AbstractIlluminaPositionFileReader;
+import picard.illumina.parser.readers.CbclReader;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parse cbcls Illumina Basecall files, and use them the to populate
+ * ClusterData objects.
+ */
+public class NewIlluminaDataProvider extends BaseIlluminaDataProvider {
```

Comment:
Maybe this should be `CbclIlluminaDataProvider`? `new` is only valid until the next file format appears...